File: Labs/experiment1/A_train_station.cpp
#include <iostream>
#include <stack>
using namespace std;

const int N = 2000001;
int a[N];

void solve() {
    int n;
    cin >> n;
    for (int i = 1; i <= n; i++) cin >> a[i];
    stack<int> st;
    int expected = 1, count = 0;
    for (int i = 1; i <= n; i++) {
        if (a[i] == expected) count++, expected++;
        else st.push(a[i]);
        while (!st.empty() && st.top() == expected) st.pop(), expected++;
    }
    while (!st.empty() && st.top() == expected) st.pop(), expected++;
    if (expected == n + 1) cout << "Yes" << endl << count << endl;
    else cout << "No" << endl;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int T;
    cin >> T;
    while (T--) solve();
    return 0;
}

File: Labs/experiment1/A_train_station.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))
    st = []
    expected = 1
    count = 0
    for x in a:
        if x == expected:
            count += 1
            expected += 1
        else:
            st.append(x)
        while st and st[-1] == expected:
            st.pop()
            expected += 1
    while st and st[-1] == expected:
        st.pop()
        expected += 1
    if expected == n + 1:
        print("Yes")
        print(count)
    else:
        print("No")

T = int(input())
for _ in range(T):
    solve()
